src/Stateless/Transition.cs

```diff
@@ -45,6 +48,11 @@ public Transition(TState source, TState destination, TTrigger trigger)
             public TTrigger Trigger { get { return _trigger; } }
 
             /// <summary>
+            /// Optional data associated with the transition.
+            /// </summary>
+            public object Tag { get { return _tag; } }
```

Comment:
I don't like to have one parameter do more than one thing, it reduces the clarity and makes it harder for a user to understand the API. So if we want to associate an Action with a trigger I prefer to have an explicit way of doing so. So let's keep Do(Action).